modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -179,14 +146,19 @@ public AnnotationsBasedDescriber(Class<?> extensionType, VersionResolver version
         this.versionResolver = versionResolver;
         typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader(extensionType.getClassLoader());
 
-        initialiseFieldDescribers();
+        intialiseFieldBasedParameterContributor();
     }
 
-    private void initialiseFieldDescribers()
+    private void intialiseFieldBasedParameterContributor()
     {
-        fieldDescribers = ImmutableList.of(new InfrastructureFieldDescriber(TlsContextFactory.class, TLS_ATTRIBUTE_NAME, typeLoader),
-                                           new InfrastructureFieldDescriber(ThreadingProfile.class, THREADING_PROFILE_ATTRIBUTE_NAME, typeLoader),
-                                           new DefaultFieldDescriber(typeLoader));
+        fielParameterContributor = ImmutableList.of((parameter, declarer) ->
```

Comment:
this approach seems overkill and quite hard to read.... Why didn't you keep the same approach but just with the differences you needed?